Enhance cleanup pass/fail and set flag for hyperconverged test-suite

dargasudarshan opened the pull request:
Signed-off-by: Sudarshan Darga <sudarshan.darga@cloudbyte.com>

<!--  Thanks for sending a pull request!  Here are some tips for you -->

**What this PR does / why we need it**:
- Enhance ansible playbooks to deploy applications on their own namespace and set flag for cleanup result.(Hyperconverged test-suite)
- Stern logger moved to prereq.yml
- Runs each application on its own namespace
- Performs apply/delete applications using modularization by passing required arguments.
- Cleanup gives out result by setting flag"Cleanup Pass" or "Cleanup Fail"

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
- @ksatchit Updated changes for 4 test playbooks. Can you please review this PR?

dargasudarshan commented:
[cassandra-lint.txt](https://github.com/openebs/openebs/files/2016259/cassandra-lint.txt)
[memleak-log.txt](https://github.com/openebs/openebs/files/2016260/memleak-log.txt)

```
Ansible APP [12:28 PM]
12:17:20 TEST: hyperconverged_cassandra_db_on_k8s, RESULT: Test Passed,Cleanup Passed

Ansible APP [1:22 PM]
13:17:30 TEST: volume_memcheck_test, RESULT: Test Passed,Cleanup Passed
```

dargasudarshan commented:
[percona-mysql-lint.txt](https://github.com/openebs/openebs/files/2021524/percona-mysql-lint.txt)

dargasudarshan commented:
[libiscsi-lint.txt](https://github.com/openebs/openebs/files/2021942/libiscsi-lint.txt)

ksatchit commented on e2e/ansible/playbooks/hyperconverged/test-k8s-memleak/test-mem.py:
```diff
@@ -7,15 +7,15 @@
 import time, os, sys
 list = []
-cmd_cntrl_name = "kubectl get pod -l openebs/controller=jiva-controller --no-headers | awk '{print $1}'"
+cmd_cntrl_name = "kubectl get pod -n memleak-test -l openebs/controller=jiva-controller --no-headers | awk '{print $1}'"
```
should this have been {{namespace}} instead of hardcoding the ns value ?

ksatchit commented on e2e/ansible/playbooks/hyperconverged/test-k8s-memleak/test-mem.py:
```diff
@@ -7,15 +7,15 @@
 lst.append("con")
 container_name = "-".join(lst)
-used_mem_process = "kubectl exec %s -c %s -- pmap -x 1 | awk ''/total'/ {print $3}'" %(cntrl_pod_name,container_name)
+used_mem_process = "kubectl exec %s -c %s -n memleak-test -- pmap -x 1 | awk ''/total'/ {print $3}'" %(cntrl_pod_name,container_name)
```
See previous comment on {{namespace}}

dargasudarshan commented:
Updated the test-mem.py by passing namespace as argument and commited the changes. 
Playbook output with latest changes:

```TASK [7b) Place data using dd] *******************************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-memleak/memleak.yml:132
changed: [localhost -> None] => {"changed": true, "cmd": "timeout 200s dd if=/dev/urandom of=/mnt/jiva/f1 bs=512k count=10000 oflag=dsync &\n timeout 200s dd if=/dev/urandom of=/mnt/jiva/f2 bs=1M count=10000 oflag=dsync &", "delta": "0:00:01.004520", "end": "2018-05-23 05:11:26.906399", "rc": 0, "start": "2018-05-23 05:11:25.901879", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [8) Copy the python script to kubeminion] ***************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-memleak/memleak.yml:140
changed: [localhost -> None] => {"changed": true, "checksum": "e4aab40f148e5c425668ce9d33f942bd81683caf", "dest": "/home/ubuntu/test-mem.py", "gid": 1000, "group": "ubuntu", "md5sum": "9c3e53ce955e8bc23531b01b712a6f3d", "mode": "0664", "owner": "ubuntu", "size": 1580, "src": "/home/ubuntu/.ansible/tmp/ansible-tmp-1527052287.06-271656123308391/source", "state": "file", "uid": 1000}

TASK [9) Run python script for memory usage validation it takes couple of minutes] ***************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-memleak/memleak.yml:146
changed: [localhost -> None] => {"changed": true, "cmd": "python test-mem.py memleak-test", "delta": "0:01:43.450052", "end": "2018-05-23 05:13:11.731489", "failed_when_result": false, "rc": 0, "start": "2018-05-23 05:11:28.281437", "stderr": "", "stderr_lines": [], "stdout": "kubectl exec pvc-9026749b-5e47-11e8-b5a4-02b983f0a4db-ctrl-6859945f4c-dbszd -c pvc-9026749b-5e47-11e8-b5a4-02b983f0a4db-ctrl-con -n memleak-test -- pmap -x 1 | awk ''/total'/ {print $3}'\n369.93359375 MB\n369.93359375 MB\n369.93359375 MB\n369.93359375 MB\n369.93359375 MB\n[369.93359375, 369.93359375, 369.93359375, 369.93359375, 369.93359375]\nPass", "stdout_lines": ["kubectl exec pvc-9026749b-5e47-11e8-b5a4-02b983f0a4db-ctrl-6859945f4c-dbszd -c pvc-9026749b-5e47-11e8-b5a4-02b983f0a4db-ctrl-con -n memleak-test -- pmap -x 1 | awk ''/total'/ {print $3}'", "369.93359375 MB", "369.93359375 MB", "369.93359375 MB", "369.93359375 MB", "369.93359375 MB", "[369.93359375, 369.93359375, 369.93359375, 369.93359375, 369.93359375]", "Pass"]}

TASK [Test Passed] *******************************************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-memleak/memleak.yml:152
ok: [localhost] => {"ansible_facts": {"flag": "Test Passed"}, "changed": false}
```
@ksatchit Can you please review the latest commit.
